File: python/58_Length_of_Last_Word.py
#!/usr/bin/python
import sys
import os


class Solution(object):
    def lengthOfLastWord(self, s):
        n = len(s)
        length, tag = 0, 1
        for i in range(n - 1, -1, -1):
            if s[i] == ' ' and tag:
                continue
            elif s[i] == ' ' and not tag:
            	break
            if s[i] != ' ':
                tag = 0
            length += 1
        
        return length
        
        


def main():
    s = Solution()
    print s.lengthOfLastWord("abc deffa  ")

if __name__ == "__main__":
    main()

File: c++/58_Length_of_Last_Word.cpp
#include <iostream>
#include <vector>
#include <map>
#include <string>

using namespace std;


class Solution {
public:
    int lengthOfLastWord(string s) {
    	int n = s.size();
    	int tag = 1;
    	int cnt = 0;
    	for (int i = n - 1; i > -1; i--) {
    		if (s[i] == ' ' and tag)
    			continue;
    		else if (s[i] != ' ' && tag)
    			tag = 0;
    		if (s[i] == ' ' && !tag)
    			break;
    		cnt++;
    	}
        return cnt;
    }
};


int main() {
	Solution s;
	cout << s.lengthOfLastWord("asd dsadasd sd ") << endl;
}
